tasks/main.yml

```diff
@@ -328,13 +328,55 @@
     __mssql_conf_setting_value: "{{ mssql_ip_address }}"
   when: mssql_ip_address is not none
 
-- name: Configure the TCP port setting
+- name: Register the previous tcpport setting
+  when:
+    - mssql_firewall_configure | bool
+    - mssql_firewall_close_previous | bool
+  shell: grep 'tcpport' {{ __mssql_conf_path }} || true
+  changed_when: false
+  register: __mssql_previous_tcp_port
+
+- name: >-
+    Open the {{ '1433' if mssql_tcp_port is none else mssql_tcp_port }} TCP port
+  when: mssql_firewall_configure | bool
+  include_role:
+    name: fedora.linux_system_roles.firewall
+  vars:
+    firewall:
+      - port: "{{ '1433' if mssql_tcp_port is none else mssql_tcp_port }}/tcp"
+        zone: public
+        state: enabled
+        permanent: true
+        runtime: true
+
+- name: >-
+    Configure the TCP port setting to
+    {{ '1433' if mssql_tcp_port is none else mssql_tcp_port }}
   include_tasks: mssql_conf_setting.yml
   vars:
     __mssql_conf_setting: "network tcpport"
     __mssql_conf_setting_value: "{{ mssql_tcp_port }}"
   when: mssql_tcp_port is not none
 
+- name: Close the previously set SQL Server TCP port if it changed
+  vars:
+    __tcpport: >-
+      {{ __mssql_previous_tcp_port.stdout | regex_search( '[1-9][0-9]{0,4}') }}
+    firewall:
+      - port: "{{ __tcpport if __tcpport else '1433' }}/tcp"
+        zone: public
```

Comment:
do you need to set the `zone` here?  If you omit the zone, does it use whatever is the default zone?  The user can then use the firewall role to set the default zone https://github.com/linux-system-roles/firewall/blob/master/tasks/main.yml#L43 (hmm - this isn't documented in the README, but it is definitely supported)

Reply:
Omitting the zone works fine.
With zone set to public, the firewall role does this: 
```
TASK [fedora.linux_system_roles.firewall : Configure firewall] *******************************************************************************************************************************
changed: [primary] => (item={'port': '1433/tcp', 'zone': 'public', 'state': 'enabled', 'permanent': True, 'runtime': True}) => {"__firewall_changed": true, "ansible_loop_var": "item", "changed": true, "item": {"permanent": true, "port": "1433/tcp", "runtime": true, "state": "enabled", "zone": "public"}}
```
And with an unset zone this:

```
TASK [fedora.linux_system_roles.firewall : Configure firewall] *******************************************************************************************************************************
changed: [primary] => (item={'port': '1433/tcp', 'state': 'enabled', 'permanent': True, 'runtime': True}) => {"__firewall_changed": true, "ansible_loop_var": "item", "changed": true, "item": {"permanent": true, "port": "1433/tcp", "runtime": true, "state": "enabled"}}
```
So the default zone is used in this case, this should be good.